src/sagemaker/jumpstart/curated_hub/curated_hub.py

```diff
@@ -113,3 +155,193 @@ def delete_model(self, model_name: str, model_version: str = "*") -> None:
     def delete(self) -> None:
         """Deletes this Curated Hub"""
         return self._sagemaker_session.delete_hub(self.hub_name)
+
+    def _is_invalid_model_list_input(self, model_list: List[Dict[str, str]]) -> bool:
+        """Determines if input args to ``sync`` is correct."""
+        for obj in model_list:
+            if not isinstance(obj.get("model_id"), str):
+                return True
+            if "version" in obj and not isinstance(obj["version"], str):
+                return True
+        return False
+
+    def sync(self, model_list: List[Dict[str, str]]):
+        """Syncs a list of JumpStart model ids and versions with a CuratedHub
+
+        Args:
+            model_list (List[Dict[str, str]]): List of `{ model_id: str, version: Optional[str] }`
+                objects that should be synced into the Hub.
+        """
+        if self._is_invalid_model_list_input(model_list):
+            raise ValueError(
+                "Model list should be a list of objects with values 'model_id',",
+                "and optional 'version'.",
+            )
+
+        # Fetch required information
+        # self._get_studio_manifest_map()
+        hub_models = self.list_models()
+
+        # Retrieve latest version of unspecified JumpStart model versions
+        model_version_list = []
+        for model in model_list:
+            version = model.get("version", "*")
+            if not version or version == "*":
+                model_specs = utils.verify_model_region_and_return_specs(
+                    model["model_id"], version, JumpStartScriptScope.INFERENCE, self.region
+                )
+                model["version"] = model_specs.version
+            model_version_list.append(model)
+
+        # Find synced JumpStart model versions in the Hub
+        js_models_in_hub = []
+        for hub_model in hub_models["HubContentSummaries"]:
+            # TODO: extract both in one pass
+            jumpstart_model_id = next(
+                (
+                    tag
+                    for tag in hub_model["search_keywords"]
+                    if tag.startswith("@jumpstart-model-id")
+                ),
+                None,
+            )
+            jumpstart_model_version = next(
+                (
+                    tag
+                    for tag in hub_model["search_keywords"]
+                    if tag.startswith("@jumpstart-model-version")
+                ),
+                None,
+            )
+
+            if jumpstart_model_id and jumpstart_model_version:
+                js_models_in_hub.append(hub_model)
```

Comment:
Should print logs or raise error in partial cases?

Reply:
Unnecessary imo. That means that someone tagged a Hub model with either "@jumpstart-model-id" or "@jumpstart-model-version" but not both

Reply:
Yeah let's add it. No reason not to be more transparent if customers enable verbose logging